mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -59,6 +59,28 @@ private[feature] trait StringIndexerBase extends Params with HasInputCol with Ha
   @Since("1.6.0")
   def getHandleInvalid: String = $(handleInvalid)
 
+  /**
+   * Param for how to order labels of string column. The first label after ordering is assigned
+   * an index of 0.
+   * Options are:
+   *   - 'freq_desc': descending order by label frequency (most frequent label assigned 0)
+   *   - 'freq_asc': ascending order by label frequency (least frequent label assigned 0)
+   *   - 'alphabet_desc': descending alphabetical order
+   *   - 'alphabet_asc': ascending alphabetical order
+   * Default is 'freq_desc'.
+   *
+   * @group param
+   */
+  @Since("2.2.0")
+  final val stringOrderType: Param[String] = new Param(this, "stringOrderType",
+    "The method used to order values of input column. " +
+      s"Supported options: ${StringIndexer.supportedStringOrderType.mkString(", ")}.",
+    (value: String) => StringIndexer.supportedStringOrderType.contains(value.toLowerCase))
+
+  /** @group getParam */
+  @Since("2.2.0")
+  def getStringOrderType: String = $(stringOrderType)
```

Comment:
I checked other ML classes. Looks like for a case-insensitive setting, we may do toLowerCase in its public API:

    def getStringOrderType: String = $(stringOrderType).toLowerCase

And you can use `getStringOrderType` below instead of `$(stringOrderType).toLowerCase` in `fit`.